pigpen-cascading/src/main/java/pigpen/cascading/SingleIterationSeq.java

```diff
@@ -9,30 +9,40 @@
 import clojure.lang.ISeq;
 
 /**
- * Sequence that allows only one iteration, but is immune to
- * memory problems resulting from holding onto the head.
+ * This sequence behaves like a normal iterator-seq up to a certain size.
+ * If the max size is exceeded, it will only allow one iteration. The objective
+ * is to become immune to memory problems resulting from holding onto the head for
+ * large collections but still support operations such as count, max, etc.
  */
 public class SingleIterationSeq extends ASeq {
+  private static int MAX_SIZE = 1000;
```

Comment:
This was an arbitrary nice round number that sounded reasonable (i.e. large enough for most use cases that require multiple passes, but small enough not to cause memory issues). I'm open to suggestions, and we could make this configurable.